api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -118,12 +118,19 @@ type FelixConfigurationSpec struct {
 	// attempts to acquire the iptables lock if it is not available. Lower values make Felix more
 	// responsive when the lock is contended, but use more CPU. [Default: 50ms]
 	IptablesLockProbeInterval *metav1.Duration `json:"iptablesLockProbeInterval,omitempty" configv1timescale:"milliseconds" confignamev1:"IptablesLockProbeIntervalMillis"`
-	// FeatureDetectOverride is used to override the feature detection.
+	// FeatureDetectOverride is used to override the environment feature detection.
 	// Values are specified in a comma separated list with no spaces, example;
 	// "SNATFullyRandom=true,MASQFullyRandom=false,RestoreSupportsLock=".
 	// "true" or "false" will force the feature, empty or omitted values are
 	// auto-detected.
 	FeatureDetectOverride string `json:"featureDetectOverride,omitempty" validate:"omitempty,keyValueList"`
+	// FeatureGate is used to enable or disable Calico features.
+	// Values are specified in a comma separated list with no spaces, example;
+	// "BPFConnectTimeLoadBalancingWorkaround=enabled,XyZ=false". This is
+	// primarily used to enable feature that is not fully production ready or
```

Comment:
"to enable features that are not fully production ready"

> helps to fix a corner case in certain environments, but is not need most of the time.

I don't think this is the right use-case for feature gates. Those should be under FeatureDetectOverride, or honestly should probably just have their own config options, right? They are GA features that vary based on platform / environment? 

I think we want to be super clear on the scope of each and the boundary between the two:

- FeatureDetectOverride: for enabling or disabling GA features that are typically auto-configured. Used when you need to do something sort of funky that our auto-detection doesn't cover.
- FeatureGates: for enabling or disabling non-GA features that are in tech-preview or may disappear in the future.

Anything other than those two categories should get its own top-level field in the FelixConfiguraiton object. My expectation is that configuration of features would move from FeatureGates to a top-level field once we decide it is a GA feature.

Reply:
sure, I think we are in agreement on this :-) Imo the gate should be for something that will be eventually always turned on once it settles.

We would introduce the gate just that the FeatureDetectOverride is used to override something that is detected on the platform.